pkg/reporter/reporter.go

```diff
@@ -99,27 +99,40 @@ func GenerateReportsFromMultipleDBs(reportTypes []ReportType, scDBs []*store.Sto
 }
 
 // GenerateReports generates reports of type HTML and Text
-func GenerateReports(testRunNames []string, reportTypes []ReportType, db store.Store) error {
-	mc := collector.NewMetricsCollector(db)
+func GenerateReports(testRunNames []string, reportTypes []ReportType, dbs []*store.StorageClassDB) error {
 	funcMap := map[ReportType]Reporter{
 		HTMLReport: &HTMLReporter{},
 		TextReport: &TextReporter{},
 	}
 	log.Infof("Started generating reports...")
 
+	invalidTestRuns := []string{}
+	// Checking availability of all the test runs in every database
 	for _, testRun := range testRunNames {
-		metricsCollection, err := mc.Collect(testRun)
-		if err != nil {
-			return err
-		}
+		testRunFound := false
+		for _, db := range dbs {
+			mc := collector.NewMetricsCollector(db.DB)
+			metricsCollection, err := mc.Collect(testRun)
+			if err != nil {
+				continue
+			}
```

Comment:
I agree with @adarsh-dell. Along with logger you can move line 131 inside this if block. This was we can remove testRunFound variable.
```suggestion
			if err != nil {
				log.Warnf("A warning message..")
				invalidTestRuns = append(invalidTestRuns, testRun)
				continue
			}
```